### PR TITLE
Alm board View based on WI States

### PR DESCRIPTION
**Pull request by SMahil**
Alm board view based on the state of Work Items
#578

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -140,7 +140,7 @@ if (process.env.ENV == 'inmemory') {
     FooterComponent,
     HeaderComponent,
     LoginComponent,
-    WorkItemSearchComponent
+    WorkItemSearchComponent,
```
Comment (vikram-raj) [inline src/app/app.module.ts]: No need of this extra comma.

File: src/app/board/board.component.scss [hunk 2]
```diff
@@ -2,38 +2,91 @@
...
-      height: em(100);
-      background: lighten($color-pf-black, 70);
-      overflow: scroll;
+.boardWrapper{
+  position: relative;
+    -webkit-box-flex: 1;
+    -webkit-flex-grow: 1;
+    -ms-flex-positive: 1;
+    flex-grow: 1;
+      -webkit-box-orient: vertical;
```
Comment (vikram-raj) [inline src/app/board/board.component.scss]: Wrong indentation.

File: src/app/board/board.component.scss [hunk 3]
```diff
@@ -2,38 +2,91 @@
...
+    padding-bottom: 10px;
+    position: absolute;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+
+    .board-lane {
+      width: 270px;
+    margin: 0 5px;
```
Comment (vikram-raj) [inline src/app/board/board.component.scss]: Wrong indentation.

File: src/app/board/board.component.html [hunk 4]
```diff
@@ -1,10 +1,47 @@
-<div id="board_topWorkItems" class="container-fluid col-md-12 col-sm-12 col-xs-12">
+<div id="board_topWorkItems" class="container-fluid col-md-12 col-sm-12 col-xs-12 boardWrapper">
   <div class='board container-cards-pf'>
```
Comment (vikram-raj) [inline src/app/board/board.component.html]: Follow one rule, either single quote or double quote. For HTML double quote is preferable.

File: src/assets/stylesheets/custom.scss [hunk 5]
```diff
@@ -70,6 +139,22 @@ $textCenter: center;
 .textCenter{ text-align: $textCenter; }
 .pointer{ cursor: pointer; }
 
+//user -gravatar
+
+.user-avatar{
+    height: em(20);
+    width: em(20);
+    img{
+        width: 100%;
```
Comment (vikram-raj) [inline src/assets/stylesheets/custom.scss]: You can use var $width100 here. Right?

File: src/app/board/board.component.html [hunk 6]
```diff
@@ -1,10 +1,47 @@
-<div id="board_topWorkItems" class="container-fluid col-md-12 col-sm-12 col-xs-12">
-  <div class='board container-cards-pf'>
-      <div class='lane' *ngFor='let lane of lanes'>
-          <div class="lane-header">{{lane.title}}</div>
-          <div [dragula]="'nested-bag'">
-            <div *ngFor='let card of lane.cards' class="card" (click)="gotoDetail(card)" id="{{'board_topWorkItemList_'+card.id}}">{{card.fields['system.title']}}</div>
+<div id="board_topWorkItems" class="container-fluid col-md-12 col-sm-12 col-xs-12 boardWrapper">
+  <div class="board container-cards-pf">
+      <!-- board columns -->
+      <div class="board-lane"" *ngFor='let lane of lanes'>
```
Comment (vikram-raj) [inline src/app/board/board.component.html]: Extra quote added here.

File: package.json [hunk 7]
```diff
@@ -79,6 +79,7 @@
     "ng2-dragula": "1.2.1",
     "ng2-dropdown": "0.0.15",
     "ng2-modal": "0.0.22",
+    "ng2bln-count-pipe": "^1.1.0",
```
Comment (nimishamukherjee) [inline package.json]: @sanbornsen is this the best way to get the count for each lane? Or can we create our own pipe?

**Comment by nimishamukherjee** [comment]
1. Create a few work items with assignees
2. Navigate to board view
3. Navigate back to list view
The list is broken.
This happens because the resolvers are not called when the board view is called.

**Comment by sanbornsen** [comment]
Things to do - 

- [ ] Bring board view under work item
- [ ] Make board view as a module with it's own routing
- [ ] Add user resolver to board routing to fetch creator and assignee details
- [ ] Get details module out of work item list and put it under work item module since it's used in both board and list. @joshuawilson @nimishamukherjee need a discussion.

**Comment by SMahil** [comment]
@sanbornsen @nimishamukherjee 

The things which are mentioned by Sudipto should be separate tasks

**Comment by sanbornsen** [comment]
@SMahil actually without these changes UI breaks currently. We can do a easy fix and merge this or we can do the proper changes. I think proper changes would be better.

**Comment by SMahil** [comment]
@sanbornsen but this will not be a small PR in that case

**Comment by sanbornsen** [comment]
@SMahil this will be little bigger than this one.

**Comment by SMahil** [comment]
@nimishamukherjee  I have raised the separate PR for User Avatar change. Once the functional changes are done, I will fix this PR

File: package.json (hunk 7 above)
Comment (sanbornsen) [inline package.json]: @nimishamukherjee @SMahil let it be there. It's a very small snippet of code and makes life easier.

File: src/app/app.module.ts [hunk 8]
```diff
@@ -88,7 +88,7 @@ if (process.env.ENV == 'inmemory') {
 @NgModule({
   imports: [
     BrowserModule,
-    BoardModule,
+    WorkItemBoardModule,
```
Comment (joshuawilson) [inline src/app/app.module.ts]: This should be alphabetical.

**Comment by joshuawilson** [comment]
The kabob menu does not appear to do anything yet the mouse changes and looks like it should bring up the menu.

File: src/app/work-item/work-item-board/work-item-board.component.html [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+                <li><a class="workItemList_Backlog">Move to Backlog</a></li>
+              </ul>
+            </aside>
+            <aside>
+              <div class="col-md-10 col-sm-10 col-xs-9">
+                <p class="multiline-truncation">{{ item.attributes['system.title'] }}</p>
+              </div>
+              <div class="col-md-2 col-sm-2 col-xs-3 no-padding user-avatar">
+                <img *ngIf="item.relationalData?.assignees?.length"
+            src="{{item.relationalData.assignees[0]?.attributes.imageURL + '&s=20'}}"
```
Comment (joshuawilson) [inline src/app/work-item/work-item-board/work-item-board.component.html]: These lines should be indented to match the *ngIf for better readability.

File: src/app/work-item/work-item-board/work-item-board.module.ts [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+import { CommonModule }       from '@angular/common';
+import { DragulaModule } from 'ng2-dragula/ng2-dragula';
+
+import { AlmIconModule } from './../../shared-component/icon/almicon.module';
+import { WorkItemBoardComponent }   from './work-item-board.component';
+import { BoardRoutingModule }   from './work-item-board-routing.module';
+//pipes
+
+import { AlmFilterBoardList } from './../../pipes/alm-board-filter.pipe';
+import { CountPipe } from 'ng2bln-count-pipe';
```
Comment (joshuawilson) [inline src/app/work-item/work-item-board/work-item-board.module.ts]: This is a style thing. We are trying to keep the external libraries at the top with angular first.

File: src/app/work-item/work-item-board/work-item-board.module.ts [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+
+
+@NgModule({
+  imports:      [ CommonModule, BoardRoutingModule, DragulaModule, AlmIconModule ],
```
Comment (joshuawilson) [inline src/app/work-item/work-item-board/work-item-board.module.ts]: These should be alphabetical.

File: package.json (hunk 7 above)
Comment (joshuawilson) [inline package.json]: @SMahil This lib should not have the ^ on it. That will auto-update the version.

**Comment by joshuawilson** [comment]
@nimishamukherjee once @SMahil resolves my comments you have my approval.

**Comment by SMahil** [comment]
@joshuawilson Kebab menu working implementation will be in separate PR

**Comment by joshuawilson** [comment]
@SMahil regarding the kabab menu, that is fine.

**Comment by nimishamukherjee** [comment]
[test]

**Comment by SMahil** [comment]
Commented code was for drag and drop. When Vikram will add drag and drop feature over there, He can verify and delete that chunk if it is not needed

**Pull request merged**